File: Examples/CSharp/Articles/UpdateActiveXComboBoxControl.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Aspose.Cells.Drawing;
using Aspose.Cells.Drawing.ActiveXControls;

namespace Aspose.Cells.Examples.CSharp.Articles
{
    public class UpdateActiveXComboBoxControl
    {
        public static void Run()
        {
            //Source directory
            string sourceDir = RunExamples.Get_SourceDirectory();

            //Output directory
            string outputDir = RunExamples.Get_OutputDirectory();

            // Create a workbook
            Workbook wb = new Workbook(sourceDir + "sampleUpdateActiveXComboBoxControl.xlsx");

            // Access first shape from first worksheet
            Shape shape = wb.Worksheets[0].Shapes[0];

            // Access ActiveX ComboBox Control and update its value
            if (shape.ActiveXControl != null)
            {
                // Access Shape ActiveX Control
                ActiveXControl c = shape.ActiveXControl;

                // Check if ActiveX Control is ComboBox Control
                if (c.Type == ControlType.ComboBox)
                {
                    // Type cast ActiveXControl into ComboBoxActiveXControl and change its value
                    ComboBoxActiveXControl comboBoxActiveX = (ComboBoxActiveXControl)c;
                    comboBoxActiveX.Value = "This is combo box control with updated value.";
                }
            }

            // Save the workbook
            wb.Save(outputDir + "outputUpdateActiveXComboBoxControl.xlsx");

            Console.WriteLine("UpdateActiveXComboBoxControl executed successfully.");
        }
    }
}

File: Examples/VisualBasic/Articles/UpdateActiveXComboBoxControl.vb

Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports Aspose.Cells.Drawing
Imports Aspose.Cells.Drawing.ActiveXControls

Namespace Articles
    Public Class UpdateActiveXComboBoxControl
        Public Shared Sub Run()
            ' ExStart:UpdateActiveXComboBoxControl
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType)

            ' Create a workbook
            Dim wb As New Workbook(dataDir & Convert.ToString("SourceFile.xlsx"))

            ' Access first shape from first worksheet
            Dim shape As Shape = wb.Worksheets(0).Shapes(0)

            ' Access ActiveX ComboBox Control and update its value
            If shape.ActiveXControl IsNot Nothing Then
                ' Access Shape ActiveX Control
                Dim c As ActiveXControl = shape.ActiveXControl

                ' Check if ActiveX Control is ComboBox Control
                If c.Type = ControlType.ComboBox Then
                    ' Type cast ActiveXControl into ComboBoxActiveXControl and change its value
                    Dim comboBoxActiveX As ComboBoxActiveXControl = DirectCast(c, ComboBoxActiveXControl)
                    comboBoxActiveX.Value = "This is combo box control with updated value."
                End If
            End If

            ' Save the workbook
            wb.Save(dataDir & Convert.ToString("OutputFile_out.xlsx"))
            ' ExEnd:UpdateActiveXComboBoxControl
        End Sub
    End Class
End Namespace
